http/client.go

```diff
@@ -8,114 +8,166 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
-	"os"
 	"path/filepath"
 	"strings"
 	"time"
 )
 
-// New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
+// ErrInvalidTransportType defines the error to be returned if the transport defined for the client is incorrect
+var ErrInvalidTransportType = errors.New("roundTripper transport invalid, should be http type")
+
+// ClientOption defines the format of the client option functions
+type ClientOption func(*http.Client) error
+
+// New creates a new http.Client with the Passed Client Options
+// that will have a custom certificate if it is loaded from the passed CA Bundle file and/or
 // directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
-func New(CABundleFile, CABundleDir string, httpTimeout time.Duration) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, "")
-}
+func New(opts ...ClientOption) (*http.Client, error) {
+	t := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
+
+	client := &http.Client{
+		Transport: t,
+	}
 
-// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
-// match the hostname of the server they are connecting to.
-func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
+	for _, opt := range opts {
+		if err := opt(client); err != nil {
+			return nil, err
+		}
+	}
+
+	return client, nil
 }
 
-func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
-	// go default http transport settings
-	t := &http.Transport{}
+// WithCABundleFile adds the CABundleFile cert to the the client's certPool
+func WithCABundleFile(CABundleFile string) ClientOption {
+	return func(c *http.Client) error {
+		if CABundleFile == "" {
+			return nil
+		}
```

Comment:
Should we return an error in this case, rather than do nothing? Seems strange to invoke the option with an invalid argument.

Reply:
good point, OTOH there's this point about allowing functions to receive no-op argument

Reply:
So you want to move the responsibility to the user of the sdk? ok, For sure I can do it.

Reply:
Done